cmd/agent/app/reporter/grpc/builder.go

```diff
@@ -97,14 +100,27 @@ func (b *ConnBuilder) CreateConnection(logger *zap.Logger) (*grpc.ClientConn, er
 		return nil, err
 	}
 
-	go func(cc *grpc.ClientConn) {
+	grpcMetrics := mFactory.Namespace(metrics.NSOptions{Name: "", Tags: map[string]string{"protocol": "grpc"}})
+
+	cm := reporter.WrapWithConnectMetrics(reporter.ConnectMetricsReporterParams{
+		Logger:         logger,
+		MetricsFactory: grpcMetrics,
+	})
+
+	go func(cc *grpc.ClientConn, cm *reporter.ConnectMetricsReporter) {
 		logger.Info("Checking connection to collector")
 		for {
 			s := cc.GetState()
+			if s != connectivity.Ready {
```

Comment:
nit: put `s == Ready` success case first.

Reply:
For compatibility and unnecessary changes in the future. Actually, I concern the gRPC official changes connectivity status constant in the future version, although seems less possible.